Fix "ABI check" that doesn't work on gcc >= 4.5, or indeed any compiler that
aborts on missing #include's.  Per http://bugs.mysql.com/bug.php?id=52514


Index: mysql-5.1.48/include/mysql.h
===================================================================
--- mysql-5.1.48.orig/include/mysql.h
+++ mysql-5.1.48/include/mysql.h
@@ -44,7 +44,9 @@ extern "C" {
 #endif
 
 #ifndef _global_h				/* If not standard header */
+#ifndef MYSQL_ABI_CHECK
 #include <sys/types.h>
+#endif
 #ifdef __LCC__
 #include <winsock2.h>				/* For windows */
 #endif
Index: mysql-5.1.48/include/mysql.h.pp
===================================================================
--- mysql-5.1.48.orig/include/mysql.h.pp
+++ mysql-5.1.48/include/mysql.h.pp
@@ -1,4 +1,3 @@
-#include <sys/types.h>
 typedef char my_bool;
 typedef int my_socket;
 #include "mysql_version.h"
Index: mysql-5.1.48/Makefile.am
===================================================================
--- mysql-5.1.48.orig/Makefile.am
+++ mysql-5.1.48/Makefile.am
@@ -275,10 +275,7 @@ TEST_PREPROCESSOR_HEADER = $(top_srcdir)
 #
 # 1) Generate preprocessor output for the files that need to
 #    be tested for abi/api changes. use -nostdinc to prevent
-#    generation of preprocessor output for system headers. This
-#    results in messages in stderr saying that these headers
-#    were not found. Redirect the stderr output to /dev/null
-#    to prevent seeing these messages.
+#    generation of preprocessor output for system headers.
 # 2) sed the output to 
 #    2.1) remove blank lines and lines that begin with "# "
 #    2.2) When gcc -E is run on the Mac OS  and solaris sparc platforms it
@@ -314,14 +311,14 @@ abi_check_all:	$(TEST_PREPROCESSOR_HEADE
 do_abi_check:
 	set -ex; \
 	for file in $(abi_headers); do \
-	         @CC@ -E -nostdinc -dI \
+	         @CC@ -E -nostdinc -dI -DMYSQL_ABI_CHECK \
 	                  -I$(top_srcdir)/include \
 	                  -I$(top_srcdir)/include/mysql \
 	                  -I$(top_srcdir)/sql \
 	                  -I$(top_builddir)/include \
 	                  -I$(top_builddir)/include/mysql \
 	                  -I$(top_builddir)/sql \
-	                                 $$file 2>/dev/null | \
+	                                 $$file | \
 	                  @SED@ -e '/^# /d' \
 	                            -e '/^[ 	]*$$/d' \
 	                            -e '/^#pragma GCC set_debug_pwd/d' \
